Задача выполнена

RecursiveCat opened the pull request.

RecursiveCat committed:
Задача выполнена

iBerbasov commented on src/main/java/school/faang/bjs2_87671/Notification.java:
```diff
@@ -0,0 +1,16 @@
+    private NotificationType type;
+    private String message;
+
+    public Notification(NotificationType type, String message) {
```
Этот конструктор - бойлер-плейт код, его можно так же сократить, используя Ломбок
@AllArgsConstructor

iBerbasov commented on src/main/java/school/faang/bjs2_87671/NotificationManager.java:
```diff
@@ -0,0 +1,18 @@
+    }
+
+    public void sendNotification(Notification notification) {
+        notificationActions.get(notification.getType()).accept(notification);
```
Тут нужна проверка на нулл или какой-то дефолтный хендлер, на случай, если не найдется в Мапе нужного по ключу

iBerbasov commented on src/main/java/school/faang/bjs2_87671/NotificationManager.java:
```diff
@@ -0,0 +1,18 @@
+
+
+public class NotificationManager {
+    Map<NotificationType, Consumer<Notification>> notificationActions = new HashMap<>();
```
1. Поля класса всегда должны быть приватными, нельзя открывать во вне доступ к состоянию класса
2. А почему actions ? Везде по тексту мы говорим "обработчик" , метод добавления называется registerHandler
Подобное разнообразие имен и терминов приводит к путанице и сложностям с восприятием кода. Это становится особенно заметно в больших классах, где точно начинаешь забывать эти связи между action-handler

Лучше придерживаться единообразия и назвать это чем-то вроде handlers, notificationHandlers, handlersMap и т.д.

RecursiveCat committed:
Добавить модификатор доступа к NotificationManager.notificationHandlers

Добавить проверку на null в NotificationManager.sendNotification

Добавить @AllArgsConstructor Notification
